.github/workflows/run-tests-and-mypy.yml

```diff
@@ -0,0 +1,41 @@
+name: Run tests and mypy
+
+on:
+  push:
+      branches: [ dev ]
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+
+jobs:
+
+  unit_tests:
+    if: |
+      github.event_name == 'push' ||
+      (github.event_name == 'pull_request' && github.event.pull_request.draft == false)
+    name: Unit testing
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
+
+    steps:
+
+        - name: Checkout main code and submodules
+          uses: actions/checkout@v2
+
+        - name: Set up Python ${{ matrix.python-version }}
+          uses: actions/setup-python@v2
+          with:
+            python-version: ${{ matrix.python-version }}
+
+        - name: Install main dependencies 
+          run: |
+            python -m pip install --upgrade pip
+            pip install .[dev] 
+        - name: Perform unit tests
+          run: |
+            pytest -vvv --cov sdk tests
+        - name: Mypy checks
```

Comment:
Open question: should these be two separate jobs ? Are later steps run when a given step fail ? i.e. if unit test fail, will the mypy checks be performed ?

Reply:
It will not be run if tests fail.
they don't have to extra jobs, but to me it makes sense?

Here you see this exact pipeline run:
https://github.com/pasqal-io/PyQ/actions/runs/3337352929/jobs/5523589364